packages/extension/src/components/pages/SetAccount/SetAccount.tsx

```diff
@@ -0,0 +1,349 @@
+  const [errorRequestRejection, setErrorRequestRejection] = useState<string>('');
+  const [isParamsMissing, setIsParamsMissing] = useState(false);
+  const [transaction, setTransaction] = useState<TransactionStatus>(TransactionStatus.Waiting);
+  const { setAccount } = useLedger();
+  const { network } = useNetwork();
+  const { estimatedFees, errorFees, difference, errorDifference } = useFees(
+    {
+      TransactionType: 'AccountSet',
+      Account: '',
+      ...(params.flags ? { Flags: params.flags } : {}),
+      ...(params.clearFlag ? { ClearFlag: params.clearFlag } : {}),
+      ...(params.domain ? { Domain: params.domain } : {}),
+      ...(params.emailHash ? { EmailHash: params.emailHash } : {}),
+      ...(params.messageKey ? { MessageKey: params.messageKey } : {}),
+      ...(params.NFTokenMinter ? { NFTokenMinter: params.NFTokenMinter } : {}),
+      ...(params.setFlag ? { SetFlag: params.setFlag } : {}),
+      ...(params.transferRate ? { TransferRate: params.transferRate } : {}),
+      ...(params.tickSize ? { TickSize: params.tickSize } : {})
+    },
+    params.fee
+  );
+  const { hasEnoughFunds, transactionStatusComponent } = useTransactionStatus({
+    isParamsMissing,
+    errorDifference,
+    network,
+    difference,
+    transaction,
+    errorRequestRejection
+  });
+
+  const createMessage = useCallback(
+    (messagePayload: {
+      hash: string | null | undefined;
+      error?: Error;
+    }): ReceiveSetAccountBackgroundMessage => {
+      const { hash, error } = messagePayload;
+
+      return {
+        app: GEM_WALLET,
+        type: 'RECEIVE_SET_ACCOUNT/V3',
+        payload: {
+          id: params.id,
+          type: ResponseType.Response,
+          result: hash
+            ? {
+                hash: hash
+              }
+            : undefined,
+          error: error ? serializeError(error) : undefined
+        }
+      };
+    },
+    [params.id]
+  );
+
+  useEffect(() => {
+    const queryString = window.location.search;
+    const urlParams = new URLSearchParams(queryString);
+    const id = Number(urlParams.get('id')) || 0;
+
+    // BaseTransaction fields
+    const {
+      fee,
+      sequence,
+      accountTxnID,
+      lastLedgerSequence,
+      memos,
+      signers,
+      sourceTag,
+      signingPubKey,
+      ticketSequence,
+      txnSignature
+    } = parseBaseParamsFromURLParams(urlParams);
+
+    // SetAccount fields
+    const flags = parseSetAccountFlags(urlParams.get('flags'));
+    const clearFlag = Number(urlParams.get('clearFlag')) || null;
+    const domain = urlParams.get('domain');
+    const emailHash = urlParams.get('emailHash');
+    const messageKey = urlParams.get('messageKey');
+    const NFTokenMinter = urlParams.get('NFTokenMinter');
+    const setFlag = Number(urlParams.get('setFlag')) || null;
+    const transferRate = Number(urlParams.get('transferRate')) || null;
+    const tickSize = Number(urlParams.get('tickSize')) || null;
+
+    if (
+      !flags &&
```

Comment:
I don't think you should have all of them, it seems that they are actually optional: https://js.xrpl.org/interfaces/AccountSet.html